torch/csrc/autograd/autograd_not_implemented_fallback.cpp

```diff
@@ -186,4 +186,121 @@ torch::CppFunction autogradNotImplementedFallback() {
   return torch::CppFunction::makeFromBoxedFunction<&autogradNotImplementedFallbackImpl>();
 }
 
+void ADInplaceOrViewFallbackImpl(const c10::OperatorHandle& op, c10::DispatchKeySet dispatch_keys, torch::jit::Stack* stack) {
+  const auto& schema = op.schema();
+  const auto& op_name = schema.operator_name().name;
+  const auto& arguments = schema.arguments();
+  const auto& returns = schema.returns();
+  const auto num_arguments = arguments.size();
+  const auto num_returns = returns.size();
+  const auto stack_start = stack->size() - num_arguments;
+  bool any_is_inplace = false;
+
+  at::Tensor aliased_input;
+
+  int64_t aliased_output_idx = -1;
+  for (const auto i : c10::irange(num_returns)) {
+    const auto& alias_info = returns[i].alias_info();
+    if (alias_info.has_value() && !alias_info->isWrite()) {
+      AT_ASSERT(
+        aliased_output_idx == -1,
+        "Expected only a single output in the operator schema to have a non-write alias annotation (i.e., 'Tensor(a)'). "
+        "Non-composite functions where multiple outputs are aliased with inputs aren't supported."
+        "Please rewrite your function as a composite function.");
+      aliased_output_idx = i;
+    }
+  }
+
+  int64_t aliased_input_idx = -1;
+  for (const auto i : c10::irange(num_arguments)) {
+    const auto& alias_info = arguments[i].alias_info();
+    if (alias_info.has_value()) {
+      if (!alias_info->isWrite()) {
+        AT_ASSERT(
+          aliased_input_idx == -1,
+          "Expected only a single input in the operator schema to have a non-write alias annotation (i.e., 'Tensor(a)'). "
+          "Non-composite functions where multiple inputs are aliased with outputs aren't supported. "
+          "Please rewrite your function as a composite function.");
+        aliased_input_idx = i;
+        const c10::IValue& aliased_input_iv = (*stack)[stack_start + i]; // get a reference to an ivalue on the stack
+        TORCH_CHECK(aliased_input_iv.isTensor());
+        // copy assignment
+        aliased_input = aliased_input_iv.toTensor();
+      } else {
+        any_is_inplace = true;
+      }
+    }
+  }
+  // NOTE: [ Limitations of ADInplaceOrView boxed kernel ]
+  // We can assume the view is always the first return, and that there will only be a single return
+  // TODO: Document the limitations of this kernel for custom op writers
+  TORCH_INTERNAL_ASSERT((aliased_input_idx == -1 && aliased_output_idx == -1) ||
+    (aliased_input_idx == 0 && aliased_output_idx == 0))
```

Comment:
An assert here probably isn't the best way to do it; rather it should be a TORCH_CHECK with a longer message saying what to do if you hit it